kubetest/gke_test.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func newGkeDeployerFactory(instance_group_prefix string) (*gkeDeployer, error) {
+	flag.Set("gke-environment", "staging")
+	if instance_group_prefix != "" {
+		flag.Set("gke-instance-group-prefix", instance_group_prefix)
+	} else {
+		flag.Set("gke-instance-group-prefix", "gke")
+	}
+
+	return newGKE(
+		"gke",
+		"project",
+		"",
+		"region",
+		"network",
+		"image",
+		"imageFamily",
+		"imageProject",
+		"cluster",
+		"sshProxyInstanceName",
+		new(string),
+		new(string),
+	)
+}
+
+func TestParseInstanceGroupsFromGCloud(t *testing.T) {
+	urlTemplate := `https://www.googleapis.com/compute/v1/projects/test-project/zones/%s/instanceGroupManagers/%s-io-default-pool-6f8ec327-grp`
```

Comment:
nit: this `-io-` part, where does it come from?

Reply:
I've removed it. I took it from some default instance group I got on GCP